library/alloc/src/sync.rs

```diff
@@ -51,8 +51,14 @@ mod tests;
 ///
 /// Going above this limit will abort your program (although not
 /// necessarily) at _exactly_ `MAX_REFCOUNT + 1` references.
+///
+/// This is a global invariant, and also applies when using a compare-exchange loop
+/// (see https://github.com/rust-lang/rust/issues/108706).
 const MAX_REFCOUNT: usize = (isize::MAX) as usize;
 
+const INTERNAL_OVERFLOW_ERROR: &str =
+    "Internal alloc::sync::Arc error. Perhaps you're leaking Weaks?";
```

Comment:
I wouldn't call this an "internal" error, since it's caused by something else (external) misusing Arc/Weak. And it's probably better to avoid addressing someone with "you", since we don't know if it's the programmer or someone else who will read the panic message.

How about something more simple, like `"weak counter overflowed"`?

Reply:
I called this internal because it's an unstable and undocumented error, and also as a programmer, I would find it a bit surprising that `Arc` "knows" that I've leaked `Weak`s.

Also, technically the counter doesn't overflow, but I guess we could pretend the counter is an `isize` and it did overflow, lol.

I'll write what you suggested.